File: src/main/java/fr/joellehuyen/AirFrance_FlightReviews/models/Flight.java
package fr.joellehuyen.AirFrance_FlightReviews.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name="flight")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Flight {
    @Id
    private String id;
    private String departureAirport;
    private String arrivalAirport;
    private LocalDateTime departureTime;
    private LocalDateTime arrivalTime;

    @ManyToOne
    @JoinColumn(name="airline_id", nullable=false)
    private Airline airline;
}

File: frontend/src/app/data/flight.ts
import { Airline} from "./airline";

export interface Flight {
  id:string;
  departureAirport:string;
  arrivalAirport:string;
  departureDate:Date;
  arrivalDate:Date;
  departureTime:string;
  arrivalTime:string;
  airline:Airline;
}

export type FlightCreateInput = Omit<Flight,'flightId' |'airline'>&{
  airlineId:string;
}
